### PR TITLE
Fix #92 stderr output

### LINKED ISSUE #92: Print logs to stderr instead of stdout
I was running `bt gitlabci > .gitlab-ci.yml` with an empty `problems.yaml`, and the command produced no output but did exit with code 1. I couldn't figure out what happened until I found that the `.gitlab-ci.yml` file contained `FATAL ERROR: Did not find any problem in problems.yaml.`

My suggestion is to print errors/warnings/logs/etc. to stderr instead of stdout to avoid situations like this. Only results of subcommands (like `bt gitlabci`, `bt sort`, and `bt tmp`) should be printed to stdout.

What do you think?

### PR DESCRIPTION
**Pull request by RagnarGrootKoerkamp**
@mpsijm

### CODE REVIEW

File: bin/constraints.py [hunk 1]
```diff
@@ -105,39 +106,41 @@ def f(cs):
     left_width = 8 + name_len + 2 * value_len
 
     print('{:^{width}}|{:^30}'.format('VALIDATORS', '      PROBLEM STATEMENT', width=left_width),
-          sep='')
+          sep='',
+          file=sys.stderr)
```
Comment (mpsijm) [inline bin/constraints.py:110]: I would say that the `constraints` command should show its output on the standard output stream, as it falls in the category of "actual 'program output' and not progress/diagnostics" (https://github.com/RagnarGrootKoerkamp/BAPCtools/issues/92#issuecomment-751534349).

File: bin/stats.py [hunk 2]
```diff
@@ -63,7 +64,7 @@ def stats(problems):
             format_string += ' {:>' + str(width + len(Fore.WHITE) + len(Style.RESET_ALL)) + '}'
 
     header = header_string.format(*headers)
-    print(Style.BRIGHT + header + Style.RESET_ALL)
+    print(Style.BRIGHT + header + Style.RESET_ALL, file=sys.stderr)
```
Comment (mpsijm) [inline bin/stats.py:67]: The output of the `stats` command would also be printed to stdout, right? (https://github.com/RagnarGrootKoerkamp/BAPCtools/issues/92#issuecomment-752060134)

File: bin/latex.py [hunk 3]
```diff
@@ -164,7 +165,7 @@ def build_latex_pdf(builddir, tex_path, problem_path=None):
     )
 
     if ret.ok is not True:
-        print(f'{Fore.RED}Failure compiling pdf:{Style.RESET_ALL}\n{ret.out}')
+        print(f'{Fore.RED}Failure compiling pdf:{Style.RESET_ALL}\n{ret.out}', file=sys.stderr)
```
Comment (mpsijm) [inline bin/latex.py:168]: Maybe some of the `print(..., file=sys.stderr)` can be replaced with calls to the logger functions of `util.py`? For example, in this case, a call to `error` could be used, and line 178 in this file could be a call to `log`.

**Review by mpsijm — CHANGES_REQUESTED** [review]
Thanks for picking this up! Luckily, BAPCtools only has 84 calls to `print`, so I also double-checked the ones that you didn't change :slightly_smiling_face:

File: bin/stats.py (hunk 2 above)
Comment (RagnarGrootKoerkamp) [reply bin/stats.py:67]: I figured there's no point currently. All stylized output (bold/colours) won't be easily parseable anyway, so there's no point in having it on stdout.

File: bin/constraints.py (hunk 1 above)
Comment (RagnarGrootKoerkamp) [reply bin/constraints.py:110]: works for me; i was in doubt for this one

File: bin/latex.py (hunk 3 above)
Comment (RagnarGrootKoerkamp) [reply bin/latex.py]: done - the remaining uses for `Fore` and `Style` all do some kind of custom styling so can't be replaced by log/error.

**Pull request merged**